### ISSUE TITLE
account empty sh_test for execution time of genrule doing its actual work

### ISSUE THREAD

**Issue by therc** [bazelbuild/bazel#6079]
### Description of the problem / feature request:

I have an empty sh_test that wraps a genrule which does the real work. Running the test takes 0.1 seconds, but that's misleading, because the genrule, which actually runs a py_binary, takes longer.

### Feature requests: what underlying problem are you trying to solve with this feature?

I want to time the actual time spent running the real test.

Perhaps the real problem is that there's no way to run a test with the flexibility of genrule (gentest?) and obviate the need for two separate rules. Whatever the problem (and solution), the current numbers are wrong. Perhaps it's a new argument to link a sh_test with the genrule for time accounting purposes; perhaps it's an improved sh_test/py_test; perhaps it's a new kind of test rule entirely written with lint-like tools in mind.

### Bugs: what's the simplest, easiest way to reproduce this bug? Please provide a minimal example if possible.

Basically run this:

https://stackoverflow.com/a/48146766

### What operating system are you running Bazel on?

Linux and Mac.

### What's the output of `bazel info release`?

This is bazel 0.16.1

###  Have you found anything relevant by searching the web?

No

**Comment by benjaminp** [comment]
Why does your example require a genrule? It seems like it should suffice to set `@local_pylint_config//:pylint` as in the `data` attribute of the `sh_test` and then invoke lint in the test.

**Comment by lberki** [comment]
This is working as intended; Bazel cannot know whether the actions that are not the test action should be accounted for the test and adding functionality to tell Bazel this doesn't sound like good design.

If you want information about actions in your build that are not a test action, use `bazel test --profile` and `bazel analyze-profile`. If you tell us why you think it's useful to split the test into a separate, we'll probably be able to suggest something better tailored to your purposes.
